src/util/http_utils.js

```diff
@@ -525,6 +525,11 @@ function check_headers(req, options) {
         throw new options.ErrorClass(options.error_bad_request);
     }
 
+    if (!_.includes([undefined, 'ExcludeIfInvalid', 'FailIfInvalid', 'RenameIfInvalid'], req.headers['azure-metadata-handling'])) {
```

Comment:
since this is tested on any S3 request, maybe better to only verify if defined 
```suggestion
    if (req.headers['azure-metadata-handling'] && !_.includes(['ExcludeIfInvalid', 'FailIfInvalid', 'RenameIfInvalid'], req.headers['azure-metadata-handling'])) {
```